### ISSUE TITLE
ConvertToMultiChannelBasedOnBratsClassesd might be wrong

### ISSUE THREAD

**Issue by guopengf** [Project-MONAI/tutorials#1355]
**Describe the bug**

This Class [ConvertToMultiChannelBasedOnBratsClassesd](https://github.com/Project-MONAI/tutorials/blob/c75cce3887c01fcea0b69e89f4d1e977fae2e782/acceleration/distributed_training/brats_training_ddp.py#L95) is used in [brats_training_ddp.py](https://github.com/Project-MONAI/tutorials/blob/c75cce3887c01fcea0b69e89f4d1e977fae2e782/acceleration/distributed_training/brats_training_ddp.py#L95) and [brats_segmentation_3d.ipynb](https://github.com/Project-MONAI/tutorials/blob/c75cce3887c01fcea0b69e89f4d1e977fae2e782/3d_segmentation/brats_segmentation_3d.ipynb). It may map the label data to wrong classes. According to its docstring,  it describes that label 2 is the GD-enhancing tumor, and label 3 is the non-enhancing tumor core. 

![image](https://user-images.githubusercontent.com/32000655/235046026-71401d72-e4d7-4818-9562-5f0d8bd853b1.png)

However, the json file in this downloaded dataset describes that label 2 is the non-enhancing tumor and label 3 is the GD-enhancing tumor.

**To Reproduce**

Download the dataset by from the [official website](https://drive.google.com/drive/folders/1HqEgzS8BV2c7xYNrZdEAnrHk7osJJ--2) or call monai.apps.DecathlonDataset(task="Task01_BrainTumour"). Find the dataset.json.
![image](https://user-images.githubusercontent.com/32000655/235044065-07cbffa3-2b4d-4f2b-9332-2cdafea28404.png)

**Expected behavior**
N/A

**Screenshots**
I also visually verified labels. It seems like json file is right. label 2 should be the non-enhancing tumor and label 3 should be the GD-enhancing tumor.
![image](https://user-images.githubusercontent.com/32000655/235044727-89e41ce1-a14f-4d4a-9e69-851bf50abd10.png)

**Environment (please complete the following information):**
 - OS
 - Python version
 - MONAI version [e.g. git commit hash]
 - CUDA/cuDNN version
 - GPU models and configuration

**Additional context**

**Comment by wyli** [comment]
agreed, it seems this version of the transform is correct https://github.com/Project-MONAI/MONAI/blob/a1c437111a511d17dcd0e04d491747f6677f4def/monai/transforms/utility/array.py#L1078-L1086 which is not needed for the dataset used in the tutorials. however we can still use this as an example of customized transforms, with revised docstrings cc @Nic-Ma @myron

**Comment by 25benjaminli** [comment]
If this is true, it seems like quite a significant problem. Not sure how it was written wrong in the monai transforms.

**Comment by KumoLiu** [comment]
Should be fixed in https://github.com/Project-MONAI/MONAI/pull/8246.
@guopengf, please help confirm, thanks!
